tests/foreman/api/test_smartproxy.py

```diff
@@ -14,14 +14,215 @@
+
+        @Assert: Proxy has the name updated
+        """
+        organizations = [
+            entities.Organization().create() for _ in range(2)]
+        newport = get_available_capsule_port()
+        with default_url_on_new_port(9090, newport) as url:
+            proxy = entities.SmartProxy(url=url).create()
+            proxy.organization = organizations
+            proxy = proxy.update(['organization'])
+            self.assertEqual(
+                {org.id for org in proxy.organization},
+                {org.id for org in organizations}
+            )
+        # Add capsule id to cleanup list
+        self.addCleanup(capsule_cleanup, proxy.id)
+
+    @skip_if_not_set('fake_capsules')
+    @run_only_on('sat')
+    @tier1
+    def test_positive_update_location(self):
+        """Proxy name update with the home proxy
+
+        @id: e08eaaa9-7c11-4cda-bbe7-6d1f7c732569
+
+        @Assert: Proxy has the name updated
+        """
+        locations = [entities.Location().create() for _ in range(2)]
+        new_port = get_available_capsule_port()
+        with default_url_on_new_port(9090, new_port) as url:
+            proxy = entities.SmartProxy(url=url).create()
+            proxy.location = locations
+            proxy = proxy.update(['location'])
+            self.assertEqual(
+                {loc.id for loc in proxy.location},
+                {loc.id for loc in locations}
+            )
+        # Add capsule id to cleanup list
+        self.addCleanup(capsule_cleanup, proxy.id)
+
+    @skip_if_not_set('fake_capsules')
+    @run_only_on('sat')
+    @tier2
+    def test_positive_refresh_features(self):
+        """Refresh smart proxy features, search for proxy by id
+
+        @id: d0237546-702e-4d1a-9212-8391295174da
+
+        @Assert: Proxy features are refreshed
+
+        @CaseLevel: Integration
+        """
+        # Since we want to run multiple commands against our fake capsule, we
+        # need the tunnel kept open in order not to allow different concurrent
+        # test to claim it. Thus we want to manage the tunnel manually.
+
+        # get an available port for our fake capsule
+        new_port = get_available_capsule_port()
+        with default_url_on_new_port(9090, new_port) as url:
+            proxy = entities.SmartProxy(url=url).create()
+            proxy.refresh()
+        # Add capsule id to cleanup list
+        self.addCleanup(capsule_cleanup, proxy.id)
+
+    @skip_if_not_set('fake_capsules')
+    @run_only_on('sat')
+    @tier1
+    def test_positive_import_puppet_classes(self):
+        """Import puppet classes from proxy
+
+        @id: 385efd1b-6146-47bf-babf-0127ce5955ed
+
+        @Assert: Puppet classes are imported from proxy
+        """
+        new_port = get_available_capsule_port()
+        with default_url_on_new_port(9090, new_port) as url:
+            proxy = entities.SmartProxy(url=url).create()
+            proxy.import_puppetclasses()
```

Comment:
did you really `@Assert: Puppet classes are imported from proxy`? or just performed action without verification